driver/conv_driver.hpp

```diff
@@ -531,6 +531,9 @@ int ConvDriver<Tgpu, Tref>::GetandSetData()
 template <typename Tgpu, typename Tref>
 int ConvDriver<Tgpu, Tref>::AddCmdLineArgs()
 {
+    inflags.AddInputFlag("in_layout", 'I', "NCHW", "Input Layout (Default=NCHW)", "string");
+    inflags.AddInputFlag("out_layout", 'O', "NCHW", "Output Layout (Default=NCHW)", "string");
+    inflags.AddInputFlag("fil_layout", 'f', "NCHW", "Input Layout (Default=NCHW)", "string");
```

Comment:
Please refer to [this comment](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/343#discussion_r474778988). Because of the reason there I decided that updating layout string to include non-NCHW is orthogonal to this review. Plus I don't feel strong about using the similar set of letters being a big issue here.

Reply:
This seems Ok as an intermediate solution. However, what is your future plans about this? I mean support for independent layouts for input, output and weights. Right now these defaults looks confusing.

I suggest opening a dedicated ticket, possibly labeled `request_for_comments`, for discussion and making/documenting design decisions.

Reply:
I'm fine with adding this as an issue but not leading this effort. I'm willing to review and provide my feedback if there is a proposed solution, but I don't know what that would be at the moment. The primary reason has been twofold:

- I have very little idea about the history and previous design decisions for all the layout related implementation. This makes me a very poor owner for making a fundamental change to this solution, just because of how widely `buffer_info.hpp` has been included in MIOpen. I could be making (or propose) a dozen of changes not foreseeing the impact to existing code base. Therefore I'd like to scope my effort to the code base that is strictly igemm infrastructure where I could offer the most help.
- From the igemm perspective, I don't mind the actual string representation of layout - any 4-letter string will do the job. This does not impact the functionality of MIOpen existing code base - making it less of a priority. Readability-wise, it is not as good as NCHW-KCYX-NKHW, but each layout does come closely with a tensor so wouldn't be tricky to handle.

Reply:
> I have very little idea about the history and previous design decisions

Nobody has I guess ;) I would say "it just happened".

> I could be making (or propose) a dozen of changes not foreseeing the impact to existing code base. 

That is why the reviewers are here. Working together, we all learn.

> Therefore I'd like to scope my effort to the code base that is strictly igemm infrastructure where  I could offer the most help.

I'll keep in mind this. However, as you can see, this is often impossible.

#509 created.

Reply:
Appreciate the feedback and for helping creating this issue. It is unfortunate that I still have couple of changes pending and I'm trying to make more progress in this. I'm trying to keep focused and have it wrapped up sooner rather than later as maintaining a working feature branch is also non-trivial. To give more details here, I'd prioritize on a functional igemm code generation, just so as to unblock the tuning implementation for igemm (Adding layout for all tensors will impact perfdb and tuna, another collaboration point with MIOpen developers.)